src/dns.rs

```diff
@@ -63,6 +72,32 @@ impl ToIpAddr for IpAddr {
     }
 }
 
+impl<T> ToIpAddrs for T
+where
+    T: ToIpAddr,
+{
+    fn to_ip_addrs(&self, dns: &mut Dns) -> Vec<IpAddr> {
+        vec![self.to_ip_addr(dns)]
+    }
+}
+
+impl ToIpAddrs for Vec<IpAddr> {
+    fn to_ip_addrs(&self, _: &mut Dns) -> Vec<IpAddr> {
+        self.clone()
+    }
+}
+
+impl ToIpAddrs for Regex {
+    fn to_ip_addrs(&self, dns: &mut Dns) -> Vec<IpAddr> {
+        #[allow(clippy::needless_collect)]
+        let hosts = dns.names.keys().cloned().collect::<Vec<_>>();
```

Comment:
Hmm, because of the `&mut`?

Reply:
Yes, I battled with this one for quite a while. There's an open issue for clippy not to produce a false positive in this situation: https://github.com/rust-lang/rust-clippy/issues/6066